workflow/controller/controller.go

```diff
@@ -1270,43 +1285,62 @@ func (wfc *WorkflowController) newPodInformer(ctx context.Context) (cache.Shared
 	return informer, nil
 }
 
-func (wfc *WorkflowController) newConfigMapInformer() (cache.SharedIndexInformer, error) {
-	indexInformer := v1.NewFilteredConfigMapInformer(wfc.kubeclientset, wfc.GetManagedNamespace(), 20*time.Minute, cache.Indexers{
+func (wfc *WorkflowController) newConfigMapInformer(ns string) (cache.SharedIndexInformer, error) {
+	indexInformer := v1.NewFilteredConfigMapInformer(wfc.kubeclientset, ns, 20*time.Minute, cache.Indexers{
 		indexes.ConfigMapLabelsIndex: indexes.ConfigMapIndexFunc,
 	}, func(opts *metav1.ListOptions) {
 		opts.LabelSelector = common.LabelKeyConfigMapType
```

Comment:
oh this is the indexer... this might be necessary then. and you already have the check in `isPluginCM()`, my bad

Reply:
wait no it _is_ the `TweakListOptionsFunc` (have to look a bit closely 😅). so this should be removed

Reply:
Some E2Es [failed](https://github.com/argoproj/argo-workflows/actions/runs/8578095350/job/23511562662?pr=11855) without this. They seemed to miss the label without this... would've thought it would've picked up all configmaps without this? 🤔 

I might be misunderstanding something here, I'm certainly not an Informer/`client-go` expert

Reply:
Oh, nvm, totally different bug; the wrong informer was being used -- fixed in [this commit](https://github.com/argoproj/argo-workflows/pull/11855/commits/191b441ec3730782ff5963e7feccfd5df9597bbc)